elfpy/data/contract_interface.py

```diff
@@ -126,46 +186,84 @@ def get_event_object(
     return (None, None)
 
 
-def get_smart_contract_read_call(contract: Contract, function_name: str, **function_args) -> dict[Any, Any]:
-    """Get a smart contract read call"""
-    # TODO: Fix this up to actually decode the ABI using web3
-    # decode ABI to get variable names
-    abi: ABI = contract.abi
-    abi_function_index = [idx for idx in range(len(abi)) if abi[idx].get("name") == function_name][0]
-    abi_outputs = abi[abi_function_index].get("outputs")
-    if not isinstance(abi_outputs, Sequence):
-        raise AssertionError("could not find outputs in the abi")
-    abi_components = abi_outputs[0].get("components")
-    if abi_components is None:
-        raise AssertionError("could not find output components in the abi")
-    return_value_keys = [component.get("name") for component in abi_components]
-    # get the callable contract function from function_name & call it
-    function: ContractFunction = contract.get_function_by_name(function_name)()
-    return_values = function.call(**function_args)
-    # associate returned values with the keys
-    assert len(return_value_keys) == len(return_values)
-    function_return_dict = dict((variable_name, info) for variable_name, info in zip(return_value_keys, return_values))
-    return function_return_dict
-
-
-def initialize_web3_with_http_provider(ethereum_node: URI | str, request_kwargs: dict | None = None) -> Web3:
-    """Initialize a Web3 instance using an HTTP provider and inject a geth Proof of Authority (poa) middleware.
-
-    Arguments
-    ---------
-    ethereum_node: URI | str
-        Address of the http provider
-    request_kwargs: dict
-        The HTTPProvider uses the python requests library for making requests.
-        If you would like to modify how requests are made,
-        you can use the request_kwargs to do so.
+def contract_function_abi_outputs(contract_abi: ABI, function_name: str) -> list[tuple[str, str]] | None:
```

Comment:
I think this is fine for now, but we gotta start writing some tests to verify desired behavior.  Time to write our own testing solidity contract!  add a TODO for this plz.